### PR TITLE
BI-544 - Add Build Scan Summary Table

### PR DESCRIPTION
**Pull request by RobiNino**

### CODE REVIEW

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/xrayScanViolationsTable/ViolationTableElement.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package org.jfrog.build.extractor.xrayScanViolationsTable;
+
+import org.apache.commons.lang.StringUtils;
+
+import java.util.Objects;
+
+/***
+ * Struct to hold a vulnerability table element.
+ */
+public class ViolationTableElement {
```
Comment (eyalbe4) [inline build-info-extractor/src/main/java/org/jfrog/build/extractor/xrayScanViolationsTable/ViolationTableElement.java:10]: What do you same we rename *ViolationTableElement* to *BuildScanTableElement*?
We can then also rename the package from *xrayScanViolationsTable* to *buildScanTable* and the *XrayViolationsTableHelper* class to *BuildScanTableHelper*.

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/xrayScanViolationsTable/XrayViolationsTableHelper.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+ * Helper for printing Xray scan results as a violations table to log.
+ */
+@SuppressWarnings("unused")
+public class XrayViolationsTableHelper {
+    private ArtifactoryXrayResponse scanResult;
+    private Map<Severity, Set<ViolationTableElement>> table;
+    private Log log;
+    private int longestDisplayName = 0;
+    private String tableFormat = "";
+    String TABLE_HEADLINE = "Xray Scan Results - Violations Summary:";
```
Comment (eyalbe4) [inline build-info-extractor/src/main/java/org/jfrog/build/extractor/xrayScanViolationsTable/XrayViolationsTableHelper.java:21]: Xray Scan Results - Violations Summary: --> Xray Scan Summary:

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/xrayScanViolationsTable/XrayViolationsTableHelper.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    /***
+     * Updates table format after longestDisplayName is known.
+     * Format aligns elements to the left.
+     * Padding on a column must be longer than the longest element in that column.
+     */
+    private void updateTableFormat() {
+        // Index (assuming 5 digits is sufficient).
+        tableFormat = "%-6s";
```
Comment (eyalbe4) [inline build-info-extractor/src/main/java/org/jfrog/build/extractor/xrayScanViolationsTable/XrayViolationsTableHelper.java:72]: A bit more efficient to use a StringBuilder for constructing the string.

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/xrayScanViolationsTable/XrayViolationsTableHelper.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    private void print() {
+        int line = 1;
+        Severity[] severities = Severity.values();
+
+        // Print table headline.
+        log.info(TABLE_HEADLINE);
+        // Print column headers.
+        printLine(TABLE_HEADERS.toArray());
+
+        // Print lines of violations by severity descending.
```
Comment (eyalbe4) [inline build-info-extractor/src/main/java/org/jfrog/build/extractor/xrayScanViolationsTable/XrayViolationsTableHelper.java:46]: ..severity descending. --> ... descending severity.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/xrayScanViolationsTable/XrayViolationsTableHelper.java (hunk 3 above)
Comment (RobiNino) [reply build-info-extractor/src/main/java/org/jfrog/build/extractor/xrayScanViolationsTable/XrayViolationsTableHelper.java:72]: Changed to one equation (StringBuilder is marked as redundant in this case).

**Pull request merged**